tests/test_graphql.py

```diff
@@ -0,0 +1,872 @@
+            (15, Name, "a"),
+            (16, Punctuation, ":"),
+            (17, Text, " "),
+            (18, String, '"'),
+            (19, String, "a"),
+            (20, String, '"'),
+            (21, Punctuation, ","),
+            (22, Text, " "),
+            (23, Name, "b"),
+            (24, Punctuation, ":"),
+            (25, Text, " "),
+            (26, Punctuation, "{"),
+            (27, Name, "c"),
+            (28, Punctuation, ":"),
+            (29, Text, " "),
+            (30, String, '"'),
+            (31, String, "c"),
+            (32, String, '"'),
+            (33, Punctuation, ","),
+            (34, Text, " "),
+            (35, Name, "d"),
+            (36, Punctuation, ":"),
+            (37, Text, " "),
+            (38, Punctuation, "{"),
+            (39, Name, "e"),
+            (40, Punctuation, ":"),
+            (41, Text, " "),
+            (42, String, '"'),
+            (43, String, "e"),
+            (44, String, '"'),
+            (45, Punctuation, "}"),
+            (46, Punctuation, "}"),
+            (47, Punctuation, "}"),
+            (48, Punctuation, ")"),
+            (49, Text, "\n"),
+            (50, Punctuation, "}"),
+            (51, Text, "\n"),
+        ]
+        assert_tokens_equal(self, text, expected)
+
+    def test_input_value_mixed(self):
+        text = """\
+        {
+          node1(arg: {a: [1, 2]})
+          node2(arg: [1, {a: "a"}])
+        }
+        """
+        expected = [
+            (0, Punctuation, "{"),
+            (1, Text, "\n  "),
+            (4, Name, "node1"),
+            (9, Punctuation, "("),
+            (10, Name, "arg"),
+            (13, Punctuation, ":"),
+            (14, Text, " "),
+            (15, Punctuation, "{"),
+            (16, Name, "a"),
+            (17, Punctuation, ":"),
+            (18, Text, " "),
+            (19, Punctuation, "["),
+            (20, Number.Integer, "1"),
+            (21, Punctuation, ","),
+            (22, Text, " "),
+            (23, Number.Integer, "2"),
+            (24, Punctuation, "]"),
+            (25, Punctuation, "}"),
+            (26, Punctuation, ")"),
+            (27, Text, "\n  "),
+            (30, Name, "node2"),
+            (35, Punctuation, "("),
+            (36, Name, "arg"),
+            (39, Punctuation, ":"),
+            (40, Text, " "),
+            (41, Punctuation, "["),
+            (42, Number.Integer, "1"),
+            (43, Punctuation, ","),
+            (44, Text, " "),
+            (45, Punctuation, "{"),
+            (46, Name, "a"),
+            (47, Punctuation, ":"),
+            (48, Text, " "),
+            (49, String, '"'),
+            (50, String, "a"),
+            (51, String, '"'),
+            (52, Punctuation, "}"),
+            (53, Punctuation, "]"),
+            (54, Punctuation, ")"),
+            (55, Text, "\n"),
+            (56, Punctuation, "}"),
+            (57, Text, "\n"),
+        ]
+        assert_tokens_equal(self, text, expected)
```

Comment:
On this file: this is a lot of test code, and generally most of it can simply go as the other tests in `tests/examplefiles/`, or better yet, `tests/snippets/` (which is a bit more practical for small test files; you could also move the other tests there). You could keep the tests that start on a specific state, but maybe there is a simple pattern that you can use to get in a value context, to keep everything in `tests/snippets/`?